cardea-server/src/main/java/ca/on/oicr/gsi/cardea/server/service/CaseService.java

```diff
@@ -149,13 +151,15 @@ public Set<ShesmuCase> getShesmuCases() {
         .collect(Collectors.toSet());
   }
 
-  private Set<String> getLimsIusIdsFor(Case kase) {
+  private Set<String> getLimsIusIdsForShesmu(Case kase) {
     return kase.getTests().stream()
         .map(test -> {
           var lqs = test.getLibraryQualifications().stream().map(Sample::getId);
           var fdls = test.getFullDepthSequencings().stream().map(Sample::getId);
           return Stream.concat(lqs, fdls)
               .filter(Objects::nonNull)
+              .filter(id -> iusPattern.matcher(id).matches()) // only keep library qualifications that are
```

Comment:
Minor optimization, but you could filter based on null sample.run before mapping to IDs instead